### PR TITLE
feat: added include_deleted to getWorkspaceByOwnerAndName

### PR DESCRIPTION
**Pull request by Kira-Pilot**
relates to #1955

I'm working on #1955: a ticket that will show deleted workspaces in our UI. 
@f0ssel made some helpful changes to `getWorkspaces` in #2095 to help support the aforementioned UI change.
However, @BrunoQuaresma recently updated the Workspace page so that it no longer uses the `getWorkspace` route, instead favoring `getWorkspaceByOwnerAndName`. 

This PR attempts to add an `include_deleted` param to `getWorkspaceByOwnerAndName` so I can continue work on #1955. I've never touched Go before so please let me know what I've broken 🧹 

P.S. It would be great to generate a type for this addition but idk how to do that.

### CODE REVIEW

File: coderd/workspaces.go
Comment (johnstcn) [inline coderd/workspaces.go:169]: Suggestion: for consistency, rename this query parameter to `deleted` (see `workspaces.go:41`)

```suggestion
		includeDeletedStr = r.URL.Query().Get("deleted")
```

**Review by johnstcn — COMMENTED** [review]
Nice! Just a couple of things:
1) I think we don't need to actually check if the `include_deleted` parameter is a valid boolean; just checking if it is equal to the string `true` should be enough here. I know this is taken from existing code though, so I'm willing to defer to others on this.
2) Would it be possible to add a unit test for this new behaviour in `workspace_test.go`? I'm up for pairing on this with you if you'd like!

File: coderd/workspaces.go
Comment (Emyrk) [inline coderd/workspaces.go:195]: I don't think this works as intended. The param here in the SQL matches for `deleted = $includeDeleted`.

So if you set this to true, the workspace will not show if the workspace is deleted. We will need to update the SQL too, or just always return deleted workspaces by default.

Comment (Emyrk) [reply coderd/workspaces.go:195]: See how it's handled here: https://github.com/coder/coder/blob/454aea7f85dcefb736789cb7e47872e62a19401c/coderd/workspaces.go#L57-L62

I think the `deleted` query param is kinda annoying as you have to know the state of the workspace to query it.

File: coderd/workspaces.go
Comment (Kira-Pilot) [reply coderd/workspaces.go:169]: @johnstcn this actually does something a little different than `deleted` (or at least, I want it to). The `deleted` for `/workspace` is not inclusive - if you add that param on a workspace that is not deleted, you get an error. Here, I want to get a record back if I pass through `include_deleted`, regardless of whether or not the record is actually deleted. Does that make sense? Happy to rename if it still seems off to you.

Comment (johnstcn) [reply coderd/workspaces.go:169]: Ah, that makes sense! Disregard then.

File: coderd/workspaces.go
Comment (Kira-Pilot) [reply coderd/workspaces.go:195]: @Emyrk Hmm, testing it in the UI seems to work.

1. if `include_deleted` is false, then we never enter this code block (because `errors.Is(err, sql.ErrNoRows)` is false
2. otherwise we enter this code block and try to query for a deleted workspace

In either case, if we get nothing back, we then go into the `httpapi.Forbidden(rw)` block below. I agree editing the SQL would also work. I just didn't know what other ramifications that would have.  

Let me know if I'm misunderstanding your comment!

File: coderd/workspaces.go
Comment (mafredri) [inline coderd/workspaces.go:190]:
```suggestion
```

It's best to avoid newlines between error and error handling, it's possible for unintended code additions to sneak in between otherwise. 😄

File: coderd/workspaces.go
Comment (mafredri) [inline coderd/workspaces.go:198]:
```suggestion
```

File: coderd/workspaces.go
Comment (mafredri) [inline coderd/workspaces.go:174]: A nice trick in Go is to avoid unnecessary variables in the scope by defining them in the scope of the `if`.

```suggestion
	includeDeleted := false
	if s := r.URL.Query().Get("include_deleted"); s != "" {
		var err error
		includeDeleted, err = strconv.ParseBool(s)
```

**Review by presleyp — APPROVED** [review]
Not sure why this tagged the frontend for review but I'm approving to unblock you on that front!

**Review by mafredri — APPROVED** [review]
Just some drive-by code style suggestions, looks good to me!

File: coderd/workspaces_test.go
Comment (johnstcn) [inline coderd/workspaces_test.go:277]: Suggestion: it's nice to keep the test names short when you're typing out that big `go test ...` command :-)

```suggestion
	t.Run("Deleted", func(t *testing.T) {
```

**Review by johnstcn — APPROVED** [review]
Looks good to me! Thanks Kira!

File: .vscode/settings.json
Comment (johnstcn) [inline .vscode/settings.json:86]: 👍

**Pull request merged**
